queries.sql

```diff
@@ -1,64 +1,42 @@
-SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
--- SELECT * from animals WHERE neutered = true;
-
-SELECT * FROM animals WHERE name != 'Squirtle';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;	
-
--- TRANSACTION
-BEGIN;
-
-UPDATE animals SET species = 'unspecified';
-
-ROLLBACK;
-
-BEGIN;
-
-UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
-UPDATE animals SET species = 'pokemon' WHERE name NOT LIKE '%mon';
-
-COMMIT;
-
-BEGIN;
-
-DELETE FROM animals;
-
-ROLLBACK;
-
-BEGIN;
-
-SAVEPOINT SP1;
-DELETE FROM animals WHERE date_of_birth > '2022-01-01';
-
-ROLLBACK TO SP1;
-
-SAVEPOINT SP2;
-UPDATE animals SET weight_kg=weight_kg * -1;
-
-ROLLBACK TO SP2;
-
-UPDATE animals SET weight_kg=weight_kg * -1;
-
-COMMIT;
-
--- SOME OTHER QUERIES
--- How many animals are there?
-SELECT count(*) AS total_animals FROM animals;
-
--- How many animals have never tried to escape?
-SELECT count(*) AS never_tried_to_escape FROM animals WHERE escape_attempts = 0;
-
--- What is the average weight of animals?
-SELECT CAST(AVG(weight_kg) AS DECIMAL(5,2)) AS average_weight FROM animals;
-
--- Who escapes the most, neutered or not neutered animals?
-SELECT name, MAX(escape_attempts) AS most_escape FROM animals WHERE neutered = true OR neutered = false GROUP BY name ORDER BY MAX(escape_attempts) DESC;
-
--- What is the minimum and maximum weight of each type of animal?
-SELECT MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals;
-
--- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
-SELECT CAST(AVG(escape_attempts) AS DECIMAL(5,2)) AS average_escape_attempts FROM animals WHERE extract(YEAR FROM date_of_birth) BETWEEN '1990' AND '2000';
+-- What animals belong to Melody Pond?
+SELECT animals.name AS animal_name, owners.full_name AS owner_name
+FROM animals
+INNER JOIN owners ON animals.owner_id = owners.id
+WHERE owners.full_name = 'Melody Pond';
+
+-- List of all animals that are Pokemon (their type is Pokemon).
+SELECT animals.name AS animal_name, species.name AS species_name
+FROM animals
+INNER JOIN species ON animals.species_id = species.id
+WHERE species.name = 'Pokemon';
+
+-- List all owners and their animals, including those who don't own any animals.
+SELECT owners.full_name AS owner_name, COALESCE(animals.name, 'No animals') AS animal_name
+FROM owners
+LEFT JOIN animals ON owners.id = animals.owner_id;
+
+-- How many animals are there per species?
+SELECT species.name AS species_name, COUNT(animals.name) AS animal_count
+FROM species
+LEFT JOIN animals ON species.id = animals.species_id
+GROUP BY species.name;
+
+-- List all Digimon owned by Jennifer Orwell.
+SELECT animals.name AS animal_name
+FROM animals
+INNER JOIN owners ON animals.owner_id = owners.id
+INNER JOIN species ON animals.species_id = species.id
+WHERE owners.full_name = 'Jennifer Orwell' AND species.name = 'Digimon';
```

Comment:
- [ ] Great job so far. However, the same error still persists with this query. You might need to check all the columns in the animals table to see which column actually exists and which is not. Doing so should put an end to this error. This is what I got  when I ran the query too.

### Error msg

![db-error](https://github.com/gilberthappi/Vet-clinic-database/assets/26597432/5497c4b2-02f6-4cc8-81c1-d9a045e3cb8f)